gocd/templates/pipelines/uptime-checker.libsonnet

```diff
@@ -33,46 +64,206 @@ local checks_stage = {
   },
 };
 
-local deploy_canary_stage(region) =
-  if region == 'us' then
-    [
-      {
-        'deploy-canary': {
-          fetch_materials: true,
-          jobs: {
-            deploy: {
-              timeout: 600,
-              elastic_profile_id: 'uptime-checker',
-              environment_variables: {
-                LABEL_SELECTOR: 'service=uptime-checker,env=canary',
-              },
-              tasks: [
-                gocdtasks.script(importstr '../bash/deploy.sh'),
-                gocdtasks.script(importstr '../bash/wait-canary.sh'),
-              ],
+// Helper stages for canary deployment
+local cleanup_canary_stage(pops) = {
+  'cleanup-canary': {
+    fetch_materials: true,
+    jobs: {
+      ['cleanup-' + r]: {
+        elastic_profile_id: 'uptime-checker',
+        environment_variables: {
+          SENTRY_REGION: r,
+        },
+        tasks: [
+          gocdtasks.script(|||
+            echo "Cleaning up any existing canary from previous runs..."
+            kubectl scale statefulset -l "service=uptime-checker,env=canary" --replicas=0 || true
+            echo "Canary cleanup complete"
+          |||),
+        ],
+      }
+      for r in pops
+    },
+  },
+};
+
+local deploy_canary_stage(pops) = {
+  'deploy-canary': {
+    fetch_materials: true,
+    jobs: {
+      ['deploy-canary-' + r]: {
+        timeout: 600,
+        elastic_profile_id: 'uptime-checker',
+        environment_variables: {
+          SENTRY_REGION: r,
+          LABEL_SELECTOR: 'service=uptime-checker,env=canary',
+        },
+        tasks: [
+          gocdtasks.script(importstr '../bash/deploy.sh'),
+        ],
+      }
+      for r in pops
+    },
+  },
+};
+
+local scale_up_canary_stage(pops, fetch_stage) = {
+  'scale-up-canary': {
+    fetch_materials: true,
+    jobs: {
+      ['scale-' + r]: {
+        elastic_profile_id: 'uptime-checker',
+        environment_variables: {
+          SENTRY_REGION: r,
+        },
+        tasks: [
+          {
+            fetch: {
+              stage: fetch_stage,
+              job: fetch_stage + '-' + r,
+              source: 'output',
+              destination: 'artifacts',
             },
           },
-        },
-      },
-    ] else [];
+          gocdtasks.script(|||
+            PROD_REPLICAS=$(kubectl get statefulset -l "service=uptime-checker,env!=canary" -o jsonpath='{.items[0].spec.replicas}')
+            echo "Scaling canary to ${PROD_REPLICAS} replicas"
+            kubectl scale statefulset -l "service=uptime-checker,env=canary" --replicas="${PROD_REPLICAS}"
```

Comment:
Is this the correct way to do this? I realize we're probably going from 0 pods to `POD_REPLICAS`, but is this going to correctly set the replica count env var?

I'm reading the docs on `kubectl scale statefulset` and it looks like it just adds pods, it doesn't restart anything

I _think_ we would need

```
kubectl rollout restart statefulset -l "service=uptime-checker,env=canary"
```

Reply:
I don't think it needs to restart anything does it? The canary should be down at this point, and we're just scaling it up.

Reply:
It might be cleaner to move this to a bash script that is called instead